src/middlewared/middlewared/plugins/pool.py

```diff
@@ -767,44 +758,48 @@ class Config:
                         'A scrub with this pool already exists'
                     )
 
-        else:
-            verrors.add(
-                f'{schema}.pool',
-                'This field is required'
-            )
-
-        month = data.get('month')
-        if not month:
-            verrors.add(
-                f'{schema}.month',
-                'This field is required'
-            )
-        elif len(month) == 12:
-            data['month'] = '*'
-        else:
-            data['month'] = ','.join(month)
+        weekdays = data.get('dayweek')
+        if weekdays and weekdays != '*':
+            try:
+                weekdays = [int(day) for day in weekdays.split(',')]
+            except ValueError:
+                verrors.add(
+                    f'{schema}.weekday',
+                    'Please specify valid week days'
+                )
+            else:
+                if len([day for day in weekdays if day not in range(1, 8)]) > 0:
+                    verrors.add(
+                        f'{schema}.weekday',
+                        'The week days should be in range of 1-7 inclusive'
+                    )
 
-        dayweek = data.get('dayweek')
-        if not dayweek:
-            verrors.add(
-                f'{schema}.dayweek',
-                'This field is required'
-            )
-        elif len(dayweek) == 7:
-            data['dayweek'] = '*'
-        else:
-            data['dayweek'] = ','.join(dayweek)
+        months = data.get('month')
+        if months and months != '*':
```

Comment:
Same here about limiting ourselves. Also, we seem to be duplicating quite a bit of work here it seems.

What do you think we make a reusable validator? Could that be worthwhile?